sky/clouds/service_catalog/data_fetchers/fetch_gcp.py

```diff
@@ -571,20 +491,47 @@ def get_tpu_df():
     return tpu_df
 
 
-if __name__ == '__main__':
-    parser = argparse.ArgumentParser()
-    parser.add_argument(
-        '--all-regions',
-        action='store_true',
-        help='Fetch all global regions, not just the U.S. ones.')
-    args = parser.parse_args()
-    region_prefix_filter = ALL_REGION_PREFIX if args.all_regions else US_REGION_PREFIX
-
-    processed_vm_df = get_vm_df(region_prefix_filter)
-    processed_gpu_df = get_gpu_df(region_prefix_filter)
-    processed_tpu_df = get_tpu_df()
-    catalog_df = pd.concat(
-        [processed_vm_df, processed_gpu_df, processed_tpu_df])
+def post_process_a2_price(catalog_df: pd.DataFrame) -> pd.DataFrame:
+    a100_df = catalog_df[catalog_df['AcceleratorName'].isin(
+        ['A100', 'A100-80GB'])]
+
+    def _deduct_a100_price(
+            row: pd.Series) -> Tuple[Optional[float], Optional[float]]:
+        instance_type = row['InstanceType']
+        if pd.isna(instance_type) or not instance_type.startswith('a2'):
+            return row['Price'], row['SpotPrice']
+
+        zone = row['AvailabilityZone']
+        a100_type = 'A100-80GB' if 'ultragpu' in instance_type else 'A100'
+        a100_count = int(instance_type.split('-')[-1][:-1])
+        a100 = a100_df[(a100_df['AcceleratorName'] == a100_type) &
+                       (a100_df['AcceleratorCount'] == a100_count) &
+                       (a100_df['AvailabilityZone'] == zone)]
+        if a100.empty:
+            # Invalid.
+            return None, None
```

Comment:
Should we `assert not a100.empty`, i.e. in what case it will be invalid?

Reply:
Another good question. I added quite a long comment in the code. PTAL.

Reply:
Thanks! The comment makes sense!